dev/com.ibm.ws.jdbc/src/com/ibm/ws/rsadapter/impl/WSManagedConnectionFactoryImpl.java

```diff
@@ -975,13 +1056,13 @@ private Object getTraceable(Object d) throws ResourceException {
     }
 
     /**
-     * Retrieve the underlying DataSource for this ManagedConnectionFactory.
+     * Retrieve the underlying DataSource or Driver for this ManagedConnectionFactory.
      * 
-     * @return the underlying DataSource.
+     * @return the underlying DataSource or Driver.
      */
-    public CommonDataSource getUnderlyingDataSource() throws SQLException
+    public Object getUnderlyingDataSourceOrDriver() throws SQLException
```

Comment:
Don't change this signature until the consequences of doing so are investigated.  I believe this is used for wrapper pattern, and I don't want to see a driver instance being returned.  I would rather see NULL returned to ensure that unwrap doesn't allow operations on the Driver